### ISSUE TITLE
Minikube and GitLab Image Registry Support

### ISSUE THREAD

**Issue by iramsey85** [kubernetes/minikube#2498]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug Report

Please provide the following details:

**Environment**:
minikube 0.24.1
Virtualbox driver
ubuntu 16.04
docker 17.12.0-ce
gitlab

**What happened**:
My objective: Configure Minikube to pull images from my local GitLab Registry. I substituted the actual registry url with "gitlab url"

**What you expected to happen**:
Expected result: with either approach, I would have expected the image to successfully pull from my gitlab registry.

**How to reproduce it** (as minimally and precisely as possible):

approach 1:
-start minikube
-from local machine docker login to gitlab with registry account
-verify creds saved in config.json on local machine
-scp config.json to /docker on minikube
-use following yaml file to start a pod
**apiVersion: v1
kind: Pod
metadata:
   name: test-image
spec:
   containers:
     - image: gitlab url/image
       name: nginx-test
       ports:
        - containerPort: 80
          protocol: TCP**

-error found: **"Failed to pull image "gitlab url/image": rpc error: code = Unknown desc = Error response from daemon: Get https://gitlab url/v2/: x509: certificate signed by unknown authority"**

approach 2: 
- delete failed pod
- stop minikube
- changed the insecure registry config in ~/.minikube/machines/minikube/config.json
- start minikube
- scp config.json to /docker on minikube
- apply previous yaml file

error found: **Failed to pull image "gitlab url/image": rpc error: code = Unknown desc = Error response from daemon: Get https://gitlab url/v2/iramsey/data-analytics/nginx/manifests/latest: denied: access forbidden**

**Anything else do we need to know**:
Nothing really. A script of some sort would be helpful as a reference.

**Comment by iramsey85** [comment]
Closing this issue.
I was able to resolve this issue with the the following fix:
1. Create a GitLab access token with read registry scope
2. Create an ImagePullSecret containing the access token user and token as credentials
3. Append the cert for my environment to  minikube >> /etc/ssl/certs/ca-certificates.crt file
4. restart the docker service on minikube
Pod deployment should be successful after this.
